src/main/java/pl/edu/pw/elka/paprykaisalami/geeruh/issues/domain/ports/IssueService.java

```diff
@@ -39,25 +43,32 @@ public Either<DomainError, Issue> get(IssueId issueId) {
     @Valid
     public Either<DomainError, Issue> create(
             ProjectCode projectCode,
+            StatusCode statusCode,
             IssueType type,
             Summary summary,
             Description description
     ) {
+        var status = statusService.get(statusCode);
+        if(status.isLeft()){
+            return Either.left(status.getLeft());
+        }
         return projectService.get(projectCode)
-                .map(p -> issueRepository.create(projectCode, type, summary, description));
+                .map(p -> issueRepository.create(projectCode, statusCode, type, summary, description));
     }
 
     @Transactional
     @Valid
     public Either<DomainError, Issue> update(
             IssueId issueId,
+            StatusCode statusCode,
             IssueType type,
             Summary summary,
             Description description
     ) {
         return issueRepository.findById(issueId).map(
                 issue -> {
                     issue.setType(type);
+                    issue.setStatusCode(statusCode);
```

Comment:
I wouldn't update status code with just with `PUT` on issues. Instead I would make an additional endpoint, say `PUT /issues/{id}/status?newStatusId={newStatusId}`.

It better expresses what the issue can do. If we would go on and add a few more attributes to issue and leave just the `PUT` method then this api would be really awkward to use. Each time we wanted to change something we would have to send this gigantic json containing all the same values with just one changed.

Also it would be more DDD'ish and less CRUD'ish, which I believe is what we are aiming for.

That would mean that you have to also split `IssueRequest` to `IssueCreateRequest` and `IssueUpdateRequest`, because you would have to set the starting status with `POST` and not be able to alter it using `POST`. Alternatively you could add url parameter `initialStatusId={...}` to `POST`.

But if you don't have time then leave it as is.

Reply:
That makes sense. Maybe I'll also contact the FE department and ask them which option they prefer more, I think they might be more used to the way I implemented it at first, but you're right - that'd be clearer.